modules/33acrossAnalyticsAdapter.js

```diff
@@ -0,0 +1,486 @@
+  }
+
+  completeAll(reason) {
+    for (let transactionId of this.#transactionsPending) {
+      this.complete(transactionId);
+    };
+
+    log.info('All remaining transactions flushed.' + (reason ? ` Reason: ${reason}` : ''));
+  }
+
+  #flushTransactions() {
+    this.#clearSendTimeout();
+    this.#transactionsPending = new Set();
+    this.#onComplete();
+  }
+
+  // gulp-eslint is using eslint 6, a version that doesn't support private method syntax
+  // eslint-disable-next-line no-dupe-class-members
+  #clearSendTimeout() {
+    return clearTimeout(this.#sendTimeoutId);
+  }
+
+  // eslint-disable-next-line no-dupe-class-members
+  #restartSendTimeout() {
+    this.#clearSendTimeout();
+
+    this.#sendTimeoutId = setTimeout(() => {
+      if (this.#sendTimeout !== 0) {
+        log.warn(`Timed out waiting for ad transactions to complete. Sending report.`);
+      }
+
+      this.#flushTransactions();
+    }, this.#sendTimeout);
+  }
+}
+
+/**
+ * initialized during `enableAnalytics`
+ */
+const locals = {
+  /** @type {Object<string, TransactionManager>} - one manager per auction */
+  transactionManagers: {},
+  /** @type {AnalyticsCache=} */
+  cache: undefined,
+  /** sets all locals to undefined */
+  reset() {
+    this.transactionManagers = {};
+    this.cache = undefined;
+  }
+}
+
+/**
+ * @typedef {Object} AnalyticsAdapter
+ * @property {function} track
+ * @property {function} enableAnalytics
+ * @property {function} disableAnalytics
+ * @property {function} [originEnableAnalytics]
+ * @property {function} [originDisableAnalytics]
+ * @property {function} [_oldEnable]
+ */
+
+/**
+ * @type {AnalyticsAdapter}
+ */
+const analyticsAdapter = Object.assign(
+  buildAdapter({ analyticsType: 'endpoint' }),
+  { track: analyticEventHandler }
+);
+
+analyticsAdapter.originEnableAnalytics = analyticsAdapter.enableAnalytics;
+analyticsAdapter.enableAnalytics = enableAnalyticsWrapper;
+
+/**
+ * @typedef {Object} AnalyticsConfig
+ * @property {string} provider - set by pbjs at module registration time
+ * @property {Object} options
+ * @property {string} options.pid - Publisher/Partner ID
+ * @property {string} [options.endpoint=DEFAULT_ENDPOINT] - Endpoint to send analytics data
+ * @property {number} [options.timeout=DEFAULT_TRANSACTION_TIMEOUT] - Timeout for sending analytics data
+ */
+
+/**
+ * @param {AnalyticsConfig} config Analytics module configuration
+ */
+function enableAnalyticsWrapper(config) {
+  const { options } = config;
+
+  const pid = options.pid;
+  if (!pid) {
+    log.error('No partnerId provided for "options.pid". No analytics will be sent.');
+
+    return;
+  }
+
+  const endpoint = calculateEndpoint(options.endpoint);
+  this.getUrl = () => endpoint;
+
+  const timeout = calculateTransactionTimeout(options.timeout);
+  this.getTimeout = () => timeout;
+
+  locals.cache = {
+    pid,
+    auctions: {},
+  };
+
+  const usPrivacy = uspDataHandler.getConsentData();
+  if (/^1[Y|N|-]{3}$/.test(usPrivacy)) {
```

Comment:
I don't think we need to validate that the usPrivacy value has the right format.. not seeing it being validated by our  modules (33x adapter & 33x UIM) nor other modules. We could safely assume it will have the right format.

Also, as an optional feedback, I think it's fine to call uspDataHandler.getConsentData() directly from `createReportFromCache` and not having to save it in the locals global var. At the end uspDataHandler.getConsentData doesn't do much besides accessing an already calculated value.